### PR TITLE
Readback depth from GPU picking

### PR DESCRIPTION
**Pull request by Wumpf**
Introduce depth readback for GPU based picking. Demonstrated using #1751 :

https://user-images.githubusercontent.com/1220815/229367020-a954f30c-a19b-489a-81b5-8e4d6ba0d9e3.mov

Overall very straight forward, except that there's heavy workarounds needed to get this working on WebGL: We read the depth texture in a shader, render it to an RgbaF32 texture, read back that and then filter out the R channel from that.
 
### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)

Browser test matrix:

* [x] Mac Firefox
* [x] Mac Chrome
* [x] Mac Safari
* [ ] Android Chrome
* [ ] Android Firefox
* [ ] iOS Safari
* [x] Windows Firefox
* [x] Windows Chrome
* [x] Windows Edge (probably redundant to Chrome anyways)
* [x] Linux Firefox
* [x] Linux Chrome

To replicate the video dump this code in the raw_mesh python file:
```
    rr.log_rigid3(
        "world/camera", parent_from_child=((20,20,-200), (0,0,0,0)), xyz="LUF"
    )
    width = 640
    height = 480
    u_cen = width / 2
    v_cen = height / 2
    f_len = (height * width) ** 0.5
    rr.log_pinhole("world/camera/image",
                      child_from_parent = [[f_len, 0,     u_cen],
                                           [0,     f_len, v_cen],
                                           [0,     0,     1  ]],
                      width = width,
                      height = height)
```

### CODE REVIEW

**Comment by teh-cmc** [comment]
Firefox & Chromium Linux :white_check_mark:

File: crates/re_viewer/src/ui/view_spatial/scene/picking.rs [hunk 1]
```diff
@@ -207,14 +207,19 @@ pub fn picking(
     if state.closest_opaque_pick.instance_path_hash == InstancePathHash::NONE {
         if let Some(gpu_picking_result) = gpu_picking_result {
             // TODO(andreas): Pick middle pixel for now. But we soon want to snap to the closest object using a bigger picking rect.
-            let rect = gpu_picking_result.rect;
-            let picked_id = gpu_picking_result.picking_data
-                [(rect.width() / 2 + (rect.height() / 2) * rect.width()) as usize];
+            let picking_rect_position = gpu_picking_result.rect.extent / 2;
```
Comment (teh-cmc) [inline crates/re_viewer/src/ui/view_spatial/scene/picking.rs:210]: I was very confused at first ~~
```suggestion
            let pos_on_picking_rect = gpu_picking_result.rect.extent / 2;

File: crates/re_renderer/src/draw_phases/picking_layer.rs [hunk 2]
```diff
@@ -26,17 +33,58 @@ pub struct PickingResult<T: 'static + Send + Sync> {
     /// Describes the area of the picking layer that was read back.
     pub rect: IntRect,
 
-    /// Picking data for the requested rectangle.
+    /// Picking id data for the requested rectangle.
+    ///
+    /// GPU internal row padding has already been removed.
+    /// Data is stored row wise, left to right, top to bottom.
```
Comment (teh-cmc) [inline crates/re_renderer/src/draw_phases/picking_layer.rs:39]: ?

**Review by teh-cmc — APPROVED** [review]
:ship:

File: crates/re_renderer/src/draw_phases/picking_layer.rs (hunk 2 above)
Comment (Wumpf) [reply crates/re_renderer/src/draw_phases/picking_layer.rs:39]: Trying to clarify, but I'm just trying to say that the pixels in there are sane 😄

Comment (teh-cmc) [reply crates/re_renderer/src/draw_phases/picking_layer.rs]: No but I mean it's not like it's raw data?

Comment (Wumpf) [reply crates/re_renderer/src/draw_phases/picking_layer.rs]: right, it's already converted 💡 
Changed it to "pixel data"

**Pull request merged**